multicluster/service-mirror/cluster_watcher.go

```diff
@@ -1021,6 +1025,11 @@ func (rcsw *RemoteClusterServiceWatcher) repairEndpoints(ctx context.Context) er
 			continue
 		}
 
+		if rcsw.isRemoteDiscovery(svc.Labels) {
```

Comment:
we can check for the the remote discovery label directly instead of conflating it with the export label.